### ISSUE TITLE
Error with bitwise operation btw numpy array and bool in align_abs

### ISSUE THREAD

**Issue by benjello** [liam2/liam2#239]
When using align_abs on the master branch, I got the following error:
````
File "/home/benjello/openfisca/liam2/liam2/alignment.py", line 81, in align_get_indices_nd
    maybe_filter &= ~leave_filter  
TypeError: ufunc 'bitwise_and' output (typecode 'l') could not be coerced to provided output parameter (typecode '?') according to the casting rule ''same_kind''
when evaluating: if(individu.age_renseigne, individu.age_renseigne, align_abs(0.0, round((10000 * distribution_nouveaux_cotisants)), filter=individu.nouveau_cotisant, leave=individu.age_renseigne))
````
that I managed to solve using the following fix:
````
    maybe_filter &= ~np.array(leave_filter)
````
I am not sure I  undertand very well what is happening but it seem to work now.
I am using numpy version 1.13.1.
Should I prepare a PR ?

**Comment by gdementen** [comment]
You could probably also (should rather?) fix it by using `leave=individu.age_renseigne > 0` in your model.

The problem seems to be you are using an integer field as a boolean filter.

**Comment by benjello** [comment]
age_renseigne is actually a boolean

**Comment by gdementen** [comment]
Well there is some integer (long actually) in there which is used as a boolean filter. Could you tell me the dtype of each of the arrays involved as seen by align_get_indices_nd?

**Comment by gdementen** [comment]
That makes sense. But if leave_filter is a python bool, that means age_renseigne is one too, right? Then I fail to see what you are trying to achieve. An align with a leave for everybody would never "select" anybody!? And in that case I don't see the point of doing an align in the first place.

**Comment by benjello** [comment]
Ok, the error is that we used 
````
  - age_renseigne: False
````
assuming that will produce a vector of False but we ended with a python bool.
Here two questions arise:
 - How can I set a complete False vector ?
 - Shouldn't be an assert testing that leave_filter is a numpy array to avoid this kind of mistakes ?

**Comment by gdementen** [comment]
Ok, now I get it. That seems a valid use, at least for testing purposes. Could you create a PR using np.asarray for both the take and leave filters? and add a test for this case in simulation.yml?
```python
if take_filter is not None:
    take_filter = np.asarray(take_filter)
    ...
if leave_filter is not None:
    leave_filter = np.asarray(leave_filter)
    ...
```

**Comment by gdementen** [comment]
closed in #241
